akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala

```diff
@@ -9,12 +9,13 @@ import akka.http.javadsl.{ model ⇒ jm }
 import akka.http.impl.util._
 import akka.http.impl.util.JavaMapping.Implicits._
 
-final case class HttpChallenge(scheme: String, realm: String,
+final case class HttpChallenge(scheme: String, token: String, realm: String,
```

Comment:
The main problem here is that we need to find a binary backward compatible solution for adding the field.

Reply:
The `realm` parameter is just one kind of `auth-param`, I think the `params` Map is a more suitable place for it, and it will be binary backward compatible.